### PR TITLE
feat: use sentry.io

### PR DESCRIPTION
**Pull request by 0dm**
base code for sentry logging

image_720

image_720

image_720

### CODE REVIEW

**Comment by 0dm** [comment]
@abrichr have any thoughts on if we should continue with sentry? It looks like [glitchtip](https://glitchtip.com) lacks some features, but we can host it on aws / elsewhere.

File: openadapt/config.py [hunk 1]
```diff
@@ -211,3 +214,10 @@ def obfuscate(val: str, pct_reveal: float = 0.1, char: str = "*") -> str:
             ):
                 val = obfuscate(val)
             logger.info(f"{key}={val}")
+
+# sentry reporting
+if REPORT_ERRORS and git.Repo(os.getcwd()).active_branch.name == "main":  # type: ignore # noqa
+    sentry_sdk.init(
+        dsn=os.getenv("SENTRY_DSN"),
```
Comment (abrichr) [inline openadapt/config.py:221]:
```suggestion
        dsn=SENTRY_DSN,
```

File: openadapt/config.py [hunk 2]
```diff
@@ -211,3 +214,10 @@ def obfuscate(val: str, pct_reveal: float = 0.1, char: str = "*") -> str:
             ):
                 val = obfuscate(val)
             logger.info(f"{key}={val}")
+
+# sentry reporting
+if REPORT_ERRORS and git.Repo(os.getcwd()).active_branch.name == "main":  # type: ignore # noqa
```
Comment (abrichr) [inline openadapt/config.py:219]:
```suggestion
if REPORT_ERRORS and git.Repo(ROOT_DIRPATH).active_branch.name == "main":  # type: ignore # noqa
```

File: openadapt/config.py [hunk 3]
```diff
@@ -15,13 +15,20 @@
...
     "CACHE_DIR_PATH": ".cache",
     "CACHE_ENABLED": True,
     "CACHE_VERBOSITY": 0,
     "DB_ECHO": False,
     "DB_FNAME": "openadapt.db",
+    "ERROR_REPORTING_ENABLED": True,
+    "ERROR_REPORTING_DSN": (
+        "https://dcf5d7889a3b4b47ae12a3af9ffcbeb7@app.glitchtip.com/3798"
+    ),
+    "ERROR_REPORTING_BRANCH": "exception-handling",
```
Comment (abrichr) [inline openadapt/config.py:31]:
```suggestion
    "ERROR_REPORTING_BRANCH": "main",
```

**Review by abrichr — APPROVED** [review]
LGTM!

**Pull request merged**